Source/NexusForever.WorldServer/Game/Entity/Bag.cs

```diff
@@ -98,12 +112,11 @@ public void Resize(int capacityChange)
         {
             if (items.Length + capacityChange < 0)
                 throw new ArgumentOutOfRangeException(nameof(capacityChange));
-            if (items.Length + capacityChange < items.Length)
-                throw new ArgumentOutOfRangeException(nameof(capacityChange));
 
             Array.Resize(ref items, items.Length + capacityChange);
+            SlotsRemaining = items.Length - items.Count(i => i != null);
```

Comment:
I don't think we need to Count the items here. We can just add `capacityChange` to SlotsRemaining.

```suggestion
SlotsRemaining += capacityChange;
```